pep_sphinx_extensions/pep_processor/parsing/pep_banner_directive.py

```diff
@@ -109,18 +109,20 @@ class CanonicalTypingSpecBanner(PEPBanner):
 
     admonition_pre_template = (
         "This PEP is a historical document. "
-        "The up-to-date, canonical spec, {link_content}, is maintained on "
-        f"the `typing specs site <{TYPING_SPEC_BASE_URL}>`__."
+        "The up-to-date, canonical specs, {link_content}, are maintained at "
+        f"the `typing specs site <{TYPING_SPEC_BASE_URL}>`__ and the CPython "
```

Comment:
Does this mean all Final typing PEPs now link to both the spec and CPython docs? And all future ones should too?

I don't think this banner is mentioned in PEPs 1 or 12, but worth a check in case something needs an update.

Reply:
> Does this mean all Final typing PEPs now link to both the spec and CPython docs? And all future ones should too?

Not _all_ of them do (e.g. there's no useful CPython link for PEP 561), but nearly all of them do, and nearly all of them will. And I think the text still reads okay for the rare PEPs like PEP 561 that only link to the typing spec? https://pep-previews--3821.org.readthedocs.build/pep-0561/

> I don't think this banner is mentioned in PEPs 1 or 12, but worth a check in case something needs an update.

It's mentioned briefly in PEP 12: https://peps.python.org/pep-0012/#canonical-documentation-and-intersphinx. Do you think I need to mention something like "remember to link to the CPython docs as well as the typing spec!" there?

Reply:
> Not _all_ of them do (e.g. there's no useful CPython link for PEP 561), but nearly all of them do, and nearly all of them will. And I think the text still reads okay for the rare PEPs like PEP 561 that only link to the typing spec? [pep-previews--3821.org.readthedocs.build/pep-0561](https://pep-previews--3821.org.readthedocs.build/pep-0561/)

Yeah, it's fine.

> It's mentioned briefly in PEP 12: [peps.python.org/pep-0012/#canonical-documentation-and-intersphinx](https://peps.python.org/pep-0012/#canonical-documentation-and-intersphinx). Do you think I need to mention something like "remember to link to the CPython docs as well as the typing spec!" there?

A brief mention would be good. We have examples for the other directives but not for `canonical-typing-spec`, so that could help. We can do this in a followup PR if you prefer.

Reply:
Added some more PEP 12 words in https://github.com/python/peps/pull/3821/commits/172dc565573e1b0c49e5b30bb40c5e83f67841a5!